port/ucontroller/nxp/lpc17xx/lpc17_hpm.c

```diff
@@ -90,22 +90,34 @@ uint8_t ipmc_hpm_prepare_comp( void )
     ipmc_page_byte_index = 0;
     ipmc_page_addr = 0;
 
+    uint8_t ret = IPMI_CC_OK;
+
     for(uint32_t i=0; i<(sizeof(ipmc_page)/sizeof(uint32_t)); i++) {
         ipmc_page[i] = 0xFFFFFFFF;
     }
 
-    /*
-     * Assumes that the bootloader has erased the flash update
-     * area. Erasing it here will lock the flash, preventing code
-     * execution until it finishes. This can take tens of ms for
-     * multiple sector erases, and for all this time no interrupt
-     * exceptions can be served.
-     *
-     * For LPC1768 devices it would cause the HPM update to fail as it
-     * wouldn't answer the ipmi request in time.
-     */
+    /* Checks flash update region integrity */
+    for(uint32_t *ptr = (uint32_t *)update_start_addr; ptr <= update_end_addr; ptr++) {
```

Comment:
ptr should be defined as `const uint32_t* ptr`, don't cast `update_start_addr` to `uint32_t*`.